bd-logger/src/builder.rs

```diff
@@ -179,6 +180,15 @@ impl LoggerBuilder {
     let (crash_monitor_tx, crash_monitor_rx) = tokio::sync::oneshot::channel();
 
     let network_quality_provider = Arc::new(SimpleNetworkQualityProvider::default());
+
+    let feature_flags_builder = FeatureFlagsBuilder::new(
+      &self.params.sdk_directory,
+      self.params.feature_flags_file_size_bytes,
+      self.params.feature_flags_high_watermark,
+    );
+    // This is only a delete and rename, so it's fast
+    feature_flags_builder.backup_previous();
```

Comment:
Since both the reader and the writer of this won't interact with this until they are started on the event loop I think we could just move this line inside
```
    let logger_future = async move {
```
and have it happen on the bg thread?